File: Ruby/Irrgarten/monster.rb
# frozen_string_literal: true
require_relative 'labyrinth_character'
module Irrgarten

  class Monster  < LabyrinthCharacter

    @@INITIAL_HEALTH = 5

    def initialize (name, intelligence, strength)
      super(name, intelligence, strength, @@INITIAL_HEALTH)
    end






    def attack()
        att = 0.0
        att = Dice.intensity(@strength)
        return att
    end


    def defend(receivedAttack)
        isDead = dead()
        if (!isDead)
          defensiveEnergy = Dice.intensity(@intelligence)
          if (defensiveEnergy<receivedAttack)
            gotWounded()
            isDead = dead()
          end
        end
        return isDead
    end

    def gotWounded()
          @health=(@health-1)
    end


    def to_s()
      "Monster{INITIAL_HEALTH=#{@@INITIAL_HEALTH}, name=#{@name}, intelligence=#{@intelligence}, strength=#{@strength}, health=#{@health}, row=#{@row}, col=#{@col}}"
    end


  end
end

File: Java/irrgarten/src/irrgarten/modelo/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten.modelo;

/**
 *
 * @author fnand
 */
public class Monster extends LabyrinthCharacter{
    
    ///////////////////////////////////////////////////////////////////
    ////////////////////          Metodos          ////////////////////
    ///////////////////////////////////////////////////////////////////
    private static final int INITIAL_HEALTH = 5;

    
    ///////////////////////////////////////////////////////////////////
    /////////////////          Constructores          /////////////////
    ///////////////////////////////////////////////////////////////////
    public Monster(String name, float intelligence, float strength) {
        super(name, intelligence, strength, INITIAL_HEALTH);
    }
    
    ///////////////////////////////////////////////////////////////////
    ////////////////////           Get's           ////////////////////
    ///////////////////////////////////////////////////////////////////

    
    ///////////////////////////////////////////////////////////////////
    ////////////////////           Set's           ////////////////////
    ///////////////////////////////////////////////////////////////////

    
    
    ///////////////////////////////////////////////////////////////////
    ////////////////////          Metodos          ////////////////////
    ///////////////////////////////////////////////////////////////////
    
    
    @Override
    public float attack(){
        float att = 0.0f;
        att = Dice.intensity(getStrength());
        return att;
    }
    
    @Override
    public boolean defend(float receivedAttack){
        boolean isDead = dead();
        if (!isDead){
            float defensiveEnergy = Dice.intensity(getIntelligence());
            if (defensiveEnergy<receivedAttack){
                gotWounded();
                isDead = dead();
            }
        }
        return isDead;
    }

    @Override
    public String toString() {
        return super.toString();
    }
    
    

    
    
   
}
